state/state.go

```diff
@@ -115,6 +116,16 @@ func (state State) MakeBlock(
 	// Fill rest of header with state data.
 	block.ChainID = state.ChainID
 
+	// Set time
+	if height == 1 {
+		block.Time = tmtime.Now()
+		if block.Time.Before(state.LastBlockTime) {
+			block.Time = state.LastBlockTime // state.LastBlockTime for height == 1 is genesis time
```

Comment:
This suggests that https://github.com/tendermint/tendermint/issues/2294 isnt fixed here? Or at least it's still possible for a proposer to make a block before their local time reaches the genesis time. In any case I think we would want the first block to happen *after* the genesis time, and ala #2294, an honest proposer shouldn't make a block until they're locally at least that time.

Reply:
Good point. I guess that the right way to address it would be not to start consensus machinery before genesis time. "Waiting" for genesis time probably shouldn't interfere with this logic.

Reply:
> Good point. I guess that the right way to address it would be not to start consensus machinery before genesis time.

this is also requested in https://github.com/tendermint/tendermint/issues/2294